samples/DevTestLabs/AzureFunctions/PowerShell/ApplyWindowsUpdateArtifact.ps1

```diff
@@ -0,0 +1,141 @@
+}
+if (-not $Request.Params.VIRTUALMACHINENAME) {
+    Write-Error "Missing [VIRTUALMACHINENAME] in the Rest API URL"
+}
+if (-not $env:ServicePrincipal_AppId) {
+    Write-Error "Missing [ServicePrincipal_AppId] in the ApplicationSettings for the Azure Function"
+}
+if (-not $env:ServicePrincipal_Key) {
+    Write-Error "Missing [ServicePrincipal_Key] in the ApplicationSettings for the Azure Function"
+}
+if (-not $env:ServicePrincipal_Tenant) {
+    Write-Error "Missing [ServicePrincipal_Tenant] in the ApplicationSettings for the Azure Function"
+}
+
+$javascriptToCloseBrowserTab = @'
+<script language='javascript' type='text/javascript'>
+    // script to close the tab after the page has loaded
+    setTimeout(function() {
+        var x = confirm('Are you sure you want to close this tab?');
+        if (x) {
+            window.open('', '_parent', '');
+            window.close();
+        }
+    }, 5000); // 5 seconds
+</script>
+'@
+
+$htmlPageTemplateResponse = @'
+<html>
+  <head>
+    <title>DevTest Lab Extensions</title>
+    <link rel='icon shortcut' href='https://azurecomcdn.azureedge.net/cvt-cc7cfb2db134841032aa0a589fb58090be0b25d70fb9a887f22a0d84adf021a9/images/icon/favicon.ico'>
+    {JAVASCRIPT}
+  </head>
+  <body>
+    <div>
+        <img alt='LabSuccessfullyUpdated.png' src='{IMAGEURL}'>
+    </div>
+  </body>
+</html>
+'@
+
+# ---------------------------------------------------
+# Setup creds and log into Azure
+# ---------------------------------------------------
+$secpasswd = ConvertTo-SecureString $env:ServicePrincipal_Key -AsPlainText -Force
+$pscreds = New-Object System.Management.Automation.PSCredential ($env:ServicePrincipal_AppId, $secpasswd)
+
+# Log into Azure with the service principal
+$login = Login-AzAccount -ServicePrincipal -Tenant $env:ServicePrincipal_Tenant -Credential $pscreds
+
+# Make sure we're looking at the right subscription
+$sub = Select-AzSubscription -SubscriptionId $Request.Params.SUBSCRIPTIONID
+
+# ---------------------------------------------------
+# Confirm subscription and lab exist
+# ---------------------------------------------------
+
+if ($sub) {
+    # Get the Virtual Machine details
+    try {
+        $vm = Get-AzResource -ResourceGroupName $Request.Params.RESOURCEGROUPNAME -ResourceType 'Microsoft.DevTestLab/labs/virtualmachines' -Name "$($Request.Params.LABNAME)/$($Request.Params.VIRTUALMACHINENAME)" -ODataQuery '$expand=Properties($expand=ComputeVM)' -ApiVersion 2016-05-15
+    }
+    catch {
+        Write-Error "Unable to find the Virtual Machine '$($Request.Params.VIRTUALMACHINENAME)' in Lab '$($Request.Params.LABNAME)'"
+    }
+
+    # Confirm the VM is running before applying the artifact
+    if ($vm -and $vm.Properties.computeVm.statuses) {
+        # Check the VM status, we should find "PowerState/running" if the VM is running
+        $running = $vm.Properties.computeVm.statuses | Where-Object {$_.code -match "PowerState/running"}
```

Comment:
Since I wasn't able to include agent status in the C# version, I think I'm going to leave it out here too...  If you have a strong preference, let me know and I can add it in just the powershell version (unless you know of a way to do it in C#, I spent awhile and couldn't find a way with the SDK).  Thanks!

Reply:
I provided a sample for C#. If you can make it work, then think about adding it here for completeness.